include/dem/dem_action_manager.h

```diff
@@ -0,0 +1,429 @@
+#ifndef dem_action_manager_h
+#define dem_action_manager_h
+
+class DEMActionManager
+{
+public:
+  /**
+   * @brief Copy constructor as a delete function to make sure it can not be
+   * copied. Will never be used.
+   *
+   * @param copy The object to be copied
+   */
+  DEMActionManager(const DEMActionManager &copy) = delete;
+
+  /**
+   * @brief Copy constructor as a delete function to make sure it can not be
+   * assigned. Will never be used.
+   *
+   * @param copy The object to be copied
+   */
+  DEMActionManager &
+  operator=(const DEMActionManager &copy) = delete;
+
+  /**
+   * @brief Getter of the unique instance of the DEMActionManager.
+   */
+  static DEMActionManager *
+  get_action_manager();
+
+  /**
+   * @brief Reset all triggers to false.
+   * TODO
+   */
+  inline void
+  reset_triggers()
+  {
+    // First mobility status identification of the CFD time step (from the
+    // velocity computed at the first DEM time step (counter = 0) of the CFD
+    // time step) The contact search is executed to make sure the mobility
+    // status of cell match the particles that are in.
+    load_balance_trigger                 = false;
+    contact_search_trigger               = mobility_status_reset_trigger;
+    clear_tangential_overlap_trigger     = false;
+    solid_object_search_trigger          = false;
+    sparse_contacts_cells_update_trigger = false;
+    read_checkpoint_trigger              = false;
+    mobility_status_reset_trigger        = false;
+  }
+
+  /**
+   * @brief Flag that there are periodic boundary in the simulation.
+   */
+  inline void
+  set_periodic_boundaries_enabling()
```

Comment:
I am not sure if it would not be better to write it the other way around like:
enable_periodic_boundaries?
However I understand that you want the set prefix so that it is clear this is a setter. I'm not sure what's the best appraoch
maybe set_periodic_boundaries_enabled so thatthis way it matches the variables name of the bool being enabled?

Reply:
Yes `set_periodic_boundaries_enabled` is way better